components/omega/test/timeStepping/TimeStepperTest.cpp

```diff
@@ -0,0 +1,314 @@
+#include "TimeStepper.h"
+#include "../ocn/OceanTestCommon.h"
+#include "AuxiliaryState.h"
+#include "DataTypes.h"
+#include "Decomp.h"
+#include "Halo.h"
+#include "HorzMesh.h"
+#include "IO.h"
+#include "IOField.h"
+#include "Logging.h"
+#include "MachEnv.h"
+#include "MetaData.h"
+#include "OceanState.h"
+#include "OmegaKokkos.h"
+#include "TendencyTerms.h"
+#include "mpi.h"
+
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+
+using namespace OMEGA;
+
+constexpr Geometry Geom   = Geometry::Planar;
+constexpr int NVertLevels = 1;
+
+struct DecayThicknessTendency {
+   void operator()(Array2DReal ThicknessTend, const OceanState *State,
+                   const AuxiliaryState *AuxState, int ThickTimeLevel,
+                   int VelTimeLevel, Real Time) const {}
+};
+
+struct DecayVelocityTendency {
+   Real Coeff = 0.5;
+
+   Real exactSolution(Real Time) { return std::exp(-Coeff * Time); }
+
+   void operator()(Array2DReal NormalVelTend, const OceanState *State,
+                   const AuxiliaryState *AuxState, int ThickTimeLevel,
+                   int VelTimeLevel, Real Time) const {
+
+      auto *Mesh                = HorzMesh::getDefault();
+      auto NVertLevels          = NormalVelTend.extent_int(1);
+      const auto &NormalVelEdge = State->NormalVelocity[VelTimeLevel];
+
+      OMEGA_SCOPE(LocCoeff, Coeff);
+
+      parallelFor(
+          {Mesh->NEdgesAll, NVertLevels}, KOKKOS_LAMBDA(int IEdge, int K) {
+             NormalVelTend(IEdge, K) -= LocCoeff * NormalVelEdge(IEdge, K);
+          });
+   }
+};
+
+int initState() {
+   int Err = 0;
+
+   auto *Mesh  = HorzMesh::getDefault();
+   auto *State = OceanState::get("TestState");
+
+   const auto &LayerThickCell = State->LayerThickness[0];
+   const auto &NormalVelEdge  = State->NormalVelocity[0];
+
+   deepCopy(LayerThickCell, 1);
+   deepCopy(NormalVelEdge, 1);
+
+   return Err;
+}
+
+int createExactSolution(Real TimeEnd) {
+   int Err = 0;
+
+   auto *DefHalo = Halo::getDefault();
+   auto *DefMesh = HorzMesh::getDefault();
+
+   auto *ExactState =
+       OceanState::create("Exact", DefMesh, DefHalo, NVertLevels, 1);
+
+   const auto &LayerThickCell = ExactState->LayerThickness[0];
+   const auto &NormalVelEdge  = ExactState->NormalVelocity[0];
+
+   deepCopy(LayerThickCell, 1);
+   deepCopy(NormalVelEdge, DecayVelocityTendency{}.exactSolution(TimeEnd));
+
+   return Err;
+}
+
+ErrorMeasures computeErrors() {
+   const auto *DefMesh = HorzMesh::getDefault();
+
+   const auto *State      = OceanState::get("TestState");
+   const auto *ExactState = OceanState::get("Exact");
+
+   const auto &NormalVelEdge      = State->NormalVelocity[0];
+   const auto &ExactNormalVelEdge = ExactState->NormalVelocity[0];
+
+   ErrorMeasures VelErrors;
+   computeErrors(VelErrors, NormalVelEdge, ExactNormalVelEdge, DefMesh, OnEdge,
+                 NVertLevels);
```

Comment:
Are we only computing errors for the `NormalVel` arrays? Would it make sense to add a check for`LayerThick` as well?

Reply:
`LayerThick` remains constant (= 1) in this test, because it the setup tries to mimic solving a single ODE.